client/src/components/CurrentWeather.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { useState, useEffect } from "react";
+import { getCityCoordinates, getWeatherForCoordinates } from "./CurrentWeatherFunctions";
+
+type CurrentWeatherProps = {
+  city?: string;
+};
+
+function useFetchData(city: string | undefined) {
+  const [data, setData] = useState({ min: null, max: null });
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState<Error | null>(null);
```

Comment:
v praxi se `null` snazi clovek vyhybat, tady by stacilo jen
```
  const [error, setError] = useState<Error>();
```

tim padem to ma hodnotu `undefined` a to neni treba specifikovat jako generic typ u useState. Je to tak kratsi a prehlednejsi. A pokud by jsi chtel "odstranit" hodnotu ze stavu, staci zavolat napr. `setError(undefined)` , coz ten muj zapis vyse dovoluje :)